src/Pages/GradeSelect/index.js

```diff
@@ -1,7 +1,314 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import styled from 'styled-components/macro';
+import SearchWidget from './SearchWidget';
+import FlightListCard from './FlightListCard';
+import FilterSelector from './FilterSelector';
 
-export default class GradeSelect extends Component {
-  render() {
-    return <div></div>;
+function GradeSelect(props) {
+  const [flightLists, setFlightLists] = useState([]);
+  const [selected, setSelected] = useState([]);
+  const [selectedDep, setSelectedDep] = useState({});
+  const [isDropdownActive, setDropdownActive] = useState(false);
+  const [scrollTop, setScrollTop] = useState(0);
+
+  useEffect(() => {
+    fetch('/datas/toData.json')
+      .then((res) => res.json())
+      .then((data) => {
+        setFlightLists(data.flightlists);
+      });
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  function handleScroll(e) {
+    const scrollTop = ('scroll', e.srcElement.scrollingElement.scrollTop);
+    setScrollTop(scrollTop);
+  }
+
+  function toggleDropdown() {
+    setDropdownActive(!isDropdownActive);
   }
+
+  function toGoNextStage(e) {
+    e.preventDefault();
+    props.history.push('/gradeselect/arrival');
+  }
+
+  function handleSelected(target) {
+    const newSelectedDep = flightLists.filter((el) => el.id === target.id);
+    setSelectedDep(...newSelectedDep);
```

Comment:
filter 자체가 이미 새로운 배열을 리턴하기 때문에 스프레드 오퍼레이터를 사용할 필요 없을 것 같습니다
```suggestion
    setSelectedDep(flightLists.filter((el) => el.id === target.id));
```

Reply:
수정하였습니다.